helix-term/src/commands/lsp.rs

```diff
@@ -178,6 +260,33 @@ pub fn workspace_symbol_picker(cx: &mut Context) {
     )
 }
 
+pub fn diagnostics_picker(cx: &mut Context) {
+    let doc = doc!(cx.editor);
+    let language_server = language_server!(cx.editor, doc);
+    if let Some(current_url) = doc.url() {
+        let offset_encoding = language_server.offset_encoding();
+        let diagnostics = cx
+            .editor
+            .diagnostics
+            .clone()
+            .into_iter()
+            .filter(|(u, _)| *u == current_url)
```

Comment:
This is costly: you're cloning everything including things that don't match. 

Why not `editor.diagnostics.entry(current_url).or_default()`?